addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/actions/EvalInputEventBasedRuleAction.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume RENARD
+ */
+package org.nuxeo.retention.actions;
+
+import static org.nuxeo.ecm.core.bulk.BulkServiceImpl.STATUS_STREAM;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.INPUT_1;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.OUTPUT_1;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.bulk.action.computation.AbstractBulkComputation;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.platform.audit.service.NXAuditEventsService;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.retention.RetentionConstants;
+import org.nuxeo.retention.adapters.Record;
+import org.nuxeo.retention.adapters.RetentionRule;
+import org.nuxeo.retention.service.RetentionManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ * Bulk action to evaluate expression on record documents with an attached event-based retention rule. Depending on the
+ * expression evaluation outcome, a determinate retention period is computed and set on the record document.
+ *
+ * @since 11.1
+ */
+public class EvalInputEventBasedRuleAction implements StreamProcessorTopology {
+
+    public static final String ACTION_NAME = "evalInputEventBasedRule";
+
+    public static final String PARAM_EVENT = "event";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                       .addComputation(EvalEventBasedRuleComputation::new,
+                               Arrays.asList(INPUT_1 + ":" + ACTION_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class EvalEventBasedRuleComputation extends AbstractBulkComputation {
+
+        static final Logger log = LogManager.getLogger(EvalEventBasedRuleComputation.class);
```

Comment:
this field is unused

Reply:
Actually it _is_ used, in the `compute` method. Moving to `private` though.